crates/ty_python_semantic/src/types/infer.rs

```diff
@@ -5692,14 +5692,16 @@ impl<'db, 'ast> TypeInferenceBuilder<'db, 'ast> {
 
             let current_file = self.file();
 
-            if let Some(name) = expr.as_name() {
-                let skip_non_global_scopes = place_table
-                    .place_id_by_name(name)
-                    .is_some_and(|symbol_id| self.skip_non_global_scopes(file_scope_id, symbol_id));
-
-                if skip_non_global_scopes {
+            let mut is_nonlocal_binding = false;
+            if let Some(name) = expr.as_name()
+                && let Some(symbol_id) = place_table.place_id_by_name(name)
```

Comment:
Probably we don't want to start using let-chains immediately and bump the MSRV to 1.88? But I figured I'd ask :)